one-off/nuke-workitem-embedded-images.sql

```diff
@@ -0,0 +1,174 @@
+create or alter function [dbo].[OidSplit] (
+    @oid varchar(200)
+)
+returns table
+as
+return
+(
+    select 
+        substring(@oid, 1, charindex(':', @oid) - 1) COLLATE Latin1_General_BIN as AssetType,
+        cast(substring(@oid, charindex(':', @oid) + 1, len(@oid)) as int) as ID
+)
+GO
+
+create or alter function [dbo].[ExtractHashesFromUrls]
+(
+    @richtextcol nvarchar(max)
+)
+returns table
+as
+return
+(
+    with recursivecte as (
+        select
+            charindex('<img', @richtextcol) as startpos,
+            charindex('>', @richtextcol, charindex('<img', @richtextcol)) as endpos,
+            1 as rownumber
+        where charindex('<img', @richtextcol) > 0
+
+        union all
+
+        select
+            charindex('<img', @richtextcol, endpos) as startpos,
+            charindex('>', @richtextcol, charindex('<img', @richtextcol, endpos)) as endpos,
+            rownumber + 1
+        from
+            recursivecte
+        where
+            charindex('<img', @richtextcol, endpos) > 0
+    )
+    select
+        substring(
+            @richtextcol,
+            startpos,
+            endpos - startpos + 1
+        ) as imgelem,
+        convert(binary(20), '0x'+ substring(
+            @richtextcol,
+            charindex('downloadblob.img/', @richtextcol, startpos) + len('downloadblob.img/'),
+            charindex('"', @richtextcol, charindex('downloadblob.img/', @richtextcol, startpos) + len('downloadblob.img/')) - charindex('downloadblob.img/', @richtextcol, startpos) - len('downloadblob.img/')
+        ),1) as hash
+    from
+        recursivecte
+    where
+        charindex('downloadblob.img/', @richtextcol, startpos) > 0
+)
+
+GO
+
+
+create or alter function [dbo].[ReplaceBase64Images] (@RichTextCol varchar(max), @BlankBase64Image varchar(max))
+returns varchar(max)
+as
+begin
+    declare @StartIndex int = charindex('<img src="data:image/', @RichTextCol);
+
+    while @StartIndex > 0
+    begin
+        set @RichTextCol = stuff(
+            @RichTextCol,
+            @StartIndex,
+            charindex('">', substring(@RichTextCol, @StartIndex, len(@RichTextCol))) + 2,
+            @BlankBase64Image
+        );
+
+        set @StartIndex = charindex('<img src="data:image/', @RichTextCol, @StartIndex + len(@BlankBase64Image));
+    end;
```

Comment:
... same with `src="data:image/` and its immediately following `"` for inline images

also, doesn't this currently nuke the entire `<img src="data:image/...">` with `@BlankBase64Image` value? does the latter contain the entire replacement `<img ...>` tag?